components/gitpod-cli/cmd/rebuild.go

```diff
@@ -162,47 +165,56 @@ func runRebuild(ctx context.Context, supervisorClient *supervisor.SupervisorClie
 	ImageBuildDuration := time.Since(imageBuildStartTime).Milliseconds()
 	event.Set("ImageBuildDuration", ImageBuildDuration)
 
-	err = TerminateExistingContainer()
+	err = TerminateExistingContainer(ctx)
 	if err != nil {
 		event.Set("ErrorCode", utils.SystemErrorCode)
 		return err
 	}
 
-	messages := []string{
-		"\n\nYou are now connected to the container",
-		"You can inspect the container and make sure the necessary tools & libraries are installed.",
-		"When you are done, just type exit to return to your Gitpod workspace\n",
-	}
-
-	welcomeMessage := strings.Join(messages, "\n")
+	welcomeMessage := strings.Join([]string{
+		"\n\nYou are now connected to the container.",
+		"Check if all tools and libraries you need are properly installed.",
+		"When you are done, type \"exit\" to return to your Gitpod workspace.\n",
+	}, "\n")
 
-	dockerRunCmd := exec.Command(
+	dockerRunCmd := exec.CommandContext(ctx,
 		dockerPath,
 		"run",
 		"--rm",
+		"-e", fmt.Sprintf("SHELL=%s", os.Getenv("SHELL")),
```

Comment:
I don't think it is correct we are interested in SHELL set in a new image, i.e. `ENV SHELL=zsh` in Dockerfile.

Reply:
Oh, the idea was not to declare SHELL in Dockerfile, but to be able to select the shell while running the command:
```
SHELL=/usr/bin/zsh gp rebuild
```

Reply:
I think a user should use `gp env set SHELL=/usr/bin/zsh && gp rebuild` or we should have `gp rebuild -e SHELL=/usr/bin/zsh` to pretest before doing do `gp env set`. But I won't introduce new options, but rather hear from users what they want. Let's keep it as an idea.

Reply:
Updated!